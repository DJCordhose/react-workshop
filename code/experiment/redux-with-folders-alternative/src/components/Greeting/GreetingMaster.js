import React from 'react';
import { connect } from 'react-redux';

import * as actions from '../../actions';
import {filterGreetings} from '../../selectors';
import {MODE_DETAIL} from '../../actions';

function GreetingMaster(props) {
    const {greetings, onAdd} = props;
    const body = greetings.map(greeting => <tr key={greeting.id}><td>{greeting.name}</td><td>{greeting.greeting}</td></tr>);
    return (
        <div>
            <table>
                <thead>
                    <tr><th>Name</th><th>Greeting</th></tr>
                </thead>
                <tbody>
                {body}
                </tbody>
            </table>
            <button className="ActionButton"
                onClick={onAdd}>
                +
            </button>
        </div>
    );
}

export default connect(
    state => ({
        greetings: filterGreetings(state.greetings, state.filter),
        mode: state.mode
    }),
    dispatch => ({
        onAdd: () => dispatch(actions.setMode(MODE_DETAIL))
    })
)(GreetingMaster);
