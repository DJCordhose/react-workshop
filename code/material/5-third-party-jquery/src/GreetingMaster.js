import React from "react";
import Spinner from "./Spinner";

const GreetingTable = ({ greetings }) => {
  return (
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Greeting</th>
        </tr>
      </thead>
      <tbody>
        {greetings.map(greeting => (
          <tr key={greeting.id}>
            <td>{greeting.name}</td>
            <td>{greeting.greeting}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default class GreetingMaster extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      limitValue: ""
    };
  }

  onLimitChange(newValue) {
    if (newValue === null || newValue === 0) {
      newValue = "";
    } else if (newValue < 1) {
      newValue = this.props.greetings.length - 1;
    } else if (newValue > this.props.greetings.length) {
      newValue = 1;
    }

    this.setState({ limitValue: newValue });
  }

  render() {
    const { greetings, onAdd } = this.props;
    const { limitValue } = this.state;

    const limitedGreetings = limitValue > 0 ? greetings.slice(0, limitValue) : greetings;

    return (
      <div>
        {/* TODO
        
        Setze hier die Spinner Komponente ein
        - Properties:
        - 'label': irgendeine Bezeichnung
        - value: limitValue aus dem State
        - onValueChange: callback-Funktion, die mit dem Ã¼bergebenen
                         neuen Wert die Methode 'this.onLimitChange'
                         aufrufen soll

        <Spinner
          . . .
        />
        
        */}

        <div>
          Showing {limitedGreetings.length} of {greetings.length} greetings
        </div>
        <GreetingTable greetings={limitedGreetings} />
        <button onClick={onAdd}>Add</button>
      </div>
    );
  }
}
