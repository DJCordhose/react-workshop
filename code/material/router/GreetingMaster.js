import React from "react";
import { Link } from "react-router";

export default function GreetingMaster(props) {
  const { greetings, onAdd } = props;
  const body = greetings.map(greeting => (
    <tr key={greeting.id}>
      <td>{greeting.name}</td>
      <td>{greeting.greeting}</td>
      <td>
        <Link to={`/greet?name=${greeting.name}&greeting=${greeting.greeting}`}>Link</Link>
      </td>
    </tr>
  ));
  return (
    <div>
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Greeting</th>
            <th>Link</th>
          </tr>
        </thead>
        <tbody>{body}</tbody>
      </table>
      <button onClick={onAdd}>Add</button>
    </div>
  );
}
