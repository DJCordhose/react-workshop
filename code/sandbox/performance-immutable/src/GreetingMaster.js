import React from "react";

export default class GreetingMaster extends React.Component {
  // 1. OPTIMIERUNG:
  // TABELLE NUR RENDERN, WENN SICH DER INHALT GEÄNDERT HAT

  // shouldComponentUpdate(nextProps) {
  //   return nextProps.greetings !== this.props.greetings;
  // }
  // componentDidUpdate() {
  //   console.log("Greeting Master Updated with " + this.props.greetings.length + "  Greetings");
  // }

  render() {
    const { greetings } = this.props;

    const body = greetings.map(greeting => <GreetingRow key={greeting.id} greeting={greeting} />);
    return (
      <div>
        <table>
          <thead>
            <tr>
              <th>Id</th>
              <th>Name</th>
              <th>Greeting</th>
            </tr>
          </thead>
          <tbody>{body}</tbody>
        </table>
      </div>
    );
  }
}

class GreetingRow extends React.Component {
  // 2. OPTIMIERUNG: NUR NEU RENDERN, WENN SICH EINE ZEILE GEÄNDERT HAT
  // shouldComponentUpdate(nextProps) {
  //   const scu = nextProps.greeting !== this.props.greeting;
  //   return scu;
  // }

  render() {
    const { greeting } = this.props;

    return (
      <tr key={greeting.id}>
        <td>{greeting.id}</td>
        <td>{greeting.name}</td>
        <td>{greeting.greeting}</td>
      </tr>
    );
  }
}
