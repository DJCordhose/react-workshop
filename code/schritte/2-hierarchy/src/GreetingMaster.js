import React from "react";

const GreetingMaster = props => {
  const { greetings, onAdd } = props;
  const body = greetings.map(greeting => (
    <tr key={greeting.id}>
      <td>{greeting.name}</td>
      <td>{greeting.greeting}</td>
    </tr>
  ));
  return (
    <div>
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Greeting</th>
          </tr>
        </thead>
        <tbody>{body}</tbody>
      </table>
      <button onClick={onAdd}>Add</button>
    </div>
  );
};

export default GreetingMaster;
