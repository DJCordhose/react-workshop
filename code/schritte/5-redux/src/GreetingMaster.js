import React from 'react';

const GreetingRow = ({greeting, onRowClicked}) => (
    <tr onClick={() => onRowClicked(greeting)}><td>{greeting.name}</td><td>{greeting.greeting}</td></tr>
);

export default function GreetingMaster(props) {
    const {greetings, onAdd, onRowClicked} = props;
    const body = greetings.map(greeting => <GreetingRow key={greeting.id} greeting={greeting} onRowClicked={onRowClicked} />);
    return (
        <div>
            <table>
                <thead>
                    <tr><th>Name</th><th>Greeting</th></tr>
                </thead>
                <tbody>
                {body}
                </tbody>
            </table>
            <button
                onClick={onAdd}>
                Add
            </button>
        </div>
    );
}
