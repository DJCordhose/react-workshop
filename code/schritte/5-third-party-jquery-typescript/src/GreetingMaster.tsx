import * as React from "react";
import Spinner from "./Spinner";
import { Greetings } from "./types";

interface GreetingTableProps {
  greetings: Greetings;
}

const GreetingTable = ({ greetings }: GreetingTableProps) => {
  return (
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Greeting</th>
        </tr>
      </thead>
      <tbody>
        {greetings.map(greeting => (
          <tr key={greeting.id}>
            <td>{greeting.name}</td>
            <td>{greeting.greeting}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

interface GreetingMasterProps {
  greetings: Greetings;

  onAdd: () => void;
}

interface GreetingMasterState {
  limitValue: string | number;
}

export default class GreetingMaster extends React.Component<GreetingMasterProps, GreetingMasterState> {
  constructor(props: GreetingMasterProps) {
    super(props);
    this.state = {
      limitValue: ""
    };
  }

  onLimitChange(newValue: string | number) {
    if (newValue === null || newValue === 0) {
      newValue = "";
    } else if (newValue < 1) {
      newValue = this.props.greetings.length - 1;
    }

    this.setState({ limitValue: newValue });
  }

  render() {
    const { greetings, onAdd } = this.props;
    const { limitValue } = this.state;

    const limitedGreetings = typeof limitValue === "number" && limitValue > 0 ? greetings.slice(0, limitValue) : greetings;

    return (
      <div>
        <Spinner
          label="Number of Greetings to show"
          value={limitValue}
          onValueChange={newValue => this.onLimitChange(newValue)}
        />

        <GreetingTable greetings={limitedGreetings} />
        <button onClick={onAdd}>Add</button>
      </div>
    );
  }
}
