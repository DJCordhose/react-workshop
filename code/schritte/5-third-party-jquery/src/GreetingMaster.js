import React from "react";
import Spinner from "./Spinner";

const GreetingTable = ({ greetings }) => {
	return (
		<table>
			<thead>
				<tr>
					<th>Name</th>
					<th>Greeting</th>
				</tr>
			</thead>
			<tbody>
				{greetings.map(greeting => (
					<tr key={greeting.id}>
						<td>{greeting.name}</td>
						<td>{greeting.greeting}</td>
					</tr>
				))}
			</tbody>
		</table>
	);
};

export default class GreetingMaster extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			limitValue: ""
		};
	}

	onLimitChange(newValue) {
		if (newValue === null || newValue === 0) {
			newValue = "";
		} else if (newValue < 1) {
			newValue = this.props.greetings.length - 1;
		}

		this.setState({ limitValue: newValue });
	}

	render() {
		const { greetings, onAdd } = this.props;
		const { limitValue } = this.state;

		const limitedGreetings = limitValue > 0 ? greetings.slice(0, limitValue) : greetings;

		return (
			<div>
				<Spinner
					label="Number of Greetings to show"
					value={limitValue}
					onValueChange={newValue => this.onLimitChange(newValue)}
				/>

				<GreetingTable greetings={limitedGreetings} />
				<button onClick={onAdd}>Add</button>
			</div>
		);
	}
}
