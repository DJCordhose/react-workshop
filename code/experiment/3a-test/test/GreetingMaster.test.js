import React from 'react';
import GreetingMaster from '../src/GreetingMaster';
import {shallow} from 'enzyme';

const someGreetings = [
    {id: 1, name: 'Klaus', greeting: 'Moin moin'},
    {id: 2, name: 'Susi', greeting: 'Hello!'}
];

test('it should render table with all greetings', () => {
    const onAddMock = jest.fn();
    const component = shallow(<GreetingMaster greetings={someGreetings} onAdd={onAddMock}/> );
    expect(component).toMatchSnapshot();
    component.find('button').simulate('click');
    expect(onAddMock.mock.calls.length).toBe(1);
});


