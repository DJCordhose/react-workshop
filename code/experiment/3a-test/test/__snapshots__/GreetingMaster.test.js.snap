// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render table with all greetings 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <table>
        <thead>
            <tr>
                <th>
                    Name
                </th>
                <th>
                    Greeting
                </th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>
                    Klaus
                </td>
                <td>
                    Moin moin
                </td>
            </tr>
            <tr>
                <td>
                    Susi
                </td>
                <td>
                    Hello!
                </td>
            </tr>
        </tbody>
    </table>
    <button
        onClick={[Function]}
    >
        Add
    </button>
</div>,
  "nodes": Array [
    <div>
      <table>
            <thead>
                  <tr>
                        <th>
                              Name
                        </th>
                        <th>
                              Greeting
                        </th>
                  </tr>
            </thead>
            <tbody>
                  <tr>
                        <td>
                              Klaus
                        </td>
                        <td>
                              Moin moin
                        </td>
                  </tr>
                  <tr>
                        <td>
                              Susi
                        </td>
                        <td>
                              Hello!
                        </td>
                  </tr>
            </tbody>
      </table>
      <button
            onClick={[Function]}
      >
            Add
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <GreetingMaster
        greetings={
                Array [
                        Object {
                          "greeting": "Moin moin",
                          "id": 1,
                          "name": "Klaus",
                        },
                        Object {
                          "greeting": "Hello!",
                          "id": 2,
                          "name": "Susi",
                        },
                      ]
        }
        onAdd={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "greetings": Array [
            Object {
              "greeting": "Moin moin",
              "id": 1,
              "name": "Klaus",
            },
            Object {
              "greeting": "Hello!",
              "id": 2,
              "name": "Susi",
            },
          ],
          "onAdd": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <table>
                    <thead>
                              <tr>
                                        <th>
                                                  Name
                                        </th>
                                        <th>
                                                  Greeting
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              <tr>
                                        <td>
                                                  Klaus
                                        </td>
                                        <td>
                                                  Moin moin
                                        </td>
                              </tr>
                              <tr>
                                        <td>
                                                  Susi
                                        </td>
                                        <td>
                                                  Hello!
                                        </td>
                              </tr>
                    </tbody>
          </table>
          <button
                    onClick={[Function]}
          >
                    Add
          </button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <table>
                    <thead>
                              <tr>
                                        <th>
                                                  Name
                                        </th>
                                        <th>
                                                  Greeting
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              <tr>
                                        <td>
                                                  Klaus
                                        </td>
                                        <td>
                                                  Moin moin
                                        </td>
                              </tr>
                              <tr>
                                        <td>
                                                  Susi
                                        </td>
                                        <td>
                                                  Hello!
                                        </td>
                              </tr>
                    </tbody>
          </table>
          <button
                    onClick={[Function]}
          >
                    Add
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GreetingMaster
    greetings={
        Array [
            Object {
              "greeting": "Moin moin",
              "id": 1,
              "name": "Klaus",
            },
            Object {
              "greeting": "Hello!",
              "id": 2,
              "name": "Susi",
            },
          ]
    }
    onAdd={[Function]}
/>,
}
`;
